joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/AstPackagePass.scala

```diff
@@ -1,28 +1,60 @@
 package io.joern.rubysrc2cpg.passes
 
 import better.files.File
-import io.joern.rubysrc2cpg.astcreation.AstCreator
 import io.joern.rubysrc2cpg.utils.{PackageContext, PackageTable}
 import io.joern.x2cpg.datastructures.Global
 import io.shiftleft.codepropertygraph.Cpg
 import io.shiftleft.passes.ConcurrentWriterCpgPass
+import org.jruby.Ruby
+import org.jruby.ast.{Colon2Node, DefnNode, Node, NodeType, ReturnNode}
 
-import scala.util.Try
+import scala.collection.mutable.ListBuffer
+import scala.jdk.CollectionConverters.CollectionHasAsScala
 
 class AstPackagePass(cpg: Cpg, tempExtDir: String, global: Global, packageTable: PackageTable, inputPath: String)
     extends ConcurrentWriterCpgPass[String](cpg) {
+
+  private val rubyInstance = Ruby.getGlobalRuntime()
   override def generateParts(): Array[String] =
     getRubyDependenciesFile(inputPath) ++ getRubyDependenciesFile(tempExtDir)
 
   override def runOnPart(diffGraph: DiffGraphBuilder, filePath: String): Unit = {
-    Try(
-      new AstCreator(
-        filePath,
-        global,
-        PackageContext(resolveModuleNameFromPath(filePath), packageTable),
-        Option(inputPath)
-      ).createAst()
-    )
+    val moduleName = resolveModuleNameFromPath(filePath)
+    try {
+      processRubyDependencyFile(filePath, moduleName)
+    } catch {
+      case ex: Exception =>
+        println(s"Error while parsing $moduleName module File ($filePath): ${ex.getMessage}")
+    }
+  }
+
+  private def fetchMethodInfoFromNode(node: Node, currentNameSpace: ListBuffer[String], moduleName: String): Unit = {
+    if (node != null) {
+      node.getNodeType match {
+        case NodeType.CLASSNODE | NodeType.MODULENODE =>
+          val childList = node.childNodes().asScala.toList
+          if (childList.nonEmpty) {
+            val classOrModuleName = childList.head.asInstanceOf[Colon2Node].getName.toString
+            currentNameSpace.addOne(classOrModuleName)
+          }
+        case NodeType.DEFNNODE =>
+          val methodName = node.asInstanceOf[DefnNode].getName.toString
+          val classPath  = currentNameSpace.mkString(":")
+          packageTable.addPackageMethod(moduleName, methodName, classPath, "<extMod>")
+        case _ =>
+      }
+      node.childNodes().forEach(childNode => fetchMethodInfoFromNode(childNode, currentNameSpace, moduleName))
+    }
+  }
+
+  private def processRubyDependencyFile(inputPath: String, moduleName: String): Unit = {
+    val currentFile = File(inputPath)
+    if (currentFile.exists) {
+      val rubyFile = new java.io.File(inputPath)
+      val rubyCode = scala.io.Source.fromFile(rubyFile).mkString
+      val rootNode = rubyInstance.parseEval(rubyCode, inputPath, null, 0)
```

Comment:
We can pass the file directly to JRuby.

```suggestion
      val rootNode = JRubyBasedParser.parseFile(inputPath)
```

Assuming we have something like this object:

```scala
package io.joern.rubysrc2cpg.parser.jruby

import org.jruby.ast.Node
import org.jruby.{Ruby, RubyInstanceConfig}

import java.io.FileInputStream

object JRubyBasedParser {
  private final val jruby = Ruby.newInstance(new RubyInstanceConfig(true))

  def parseFile(filename: String): Node =
    jruby.parseFile(new FileInputStream(filename), filename, jruby.getCurrentContext.getCurrentScope)

  def parseString(code: String, filename: String): Node = {
    jruby.parse(code, filename, jruby.getCurrentContext.getCurrentScope, 1, true)
  }
}
```

Reply:
done.